src/config.rs

```diff
@@ -66,58 +78,48 @@ impl NetworkConfig {
     }
 }
 
-#[cfg(test)]
-impl Default for NetworkConfig {
-    fn default() -> Self {
-        NetworkConfig {
-            peers: [
-                PeerConfig::new("localhost:3000".parse().unwrap()),
-                PeerConfig::new("localhost:3001".parse().unwrap()),
-                PeerConfig::new("localhost:3002".parse().unwrap()),
-            ],
-        }
-    }
-}
-
-#[derive(Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Deserialize)]
 pub struct PeerConfig {
     /// Peer URL
     #[serde(with = "crate::uri")]
     pub url: Uri,
 
-    /// Peer's TLS certificate or CA in PEM format
+    /// Peer's TLS certificate
     ///
-    /// If the peer's TLS certificate can be verified using the system truststore, this may be omitted.
+    /// In `network.toml`, the certificate must be in PEM format. It is converted to DER
+    /// when the config is loaded.
     ///
-    /// If the peer's TLS certificate cannot be verified using the system truststore, or for a stronger
-    /// check that the peer uses the expected PKI, either the peer certificate or the authority
-    /// certificate that issues the peer's certificate may be specified here.
+    /// Either the peer's TLS certificate, or the authority certificate that issues the peer
+    /// certificate, must be specified here, unless HTTPS is disabled.
     ///
-    /// If a certificate is specified here, only the specified certificate will be accepted. The system
-    /// truststore will not be used.
-    pub certificate: Option<String>,
+    /// It is possible to rely on the system truststore and omit this configuration if the peers use
+    /// certificates from public CAs, but the client certificate verification is not currently
+    /// configured to work off of the system truststore. (Besides the mechanics of accessing the
+    /// system truststore, there is also the issue of what client CA names the server should send
+    /// in that case.)
+    #[serde(default, deserialize_with = "certificate_from_pem")]
+    pub certificate: Option<Certificate>,
```

Comment:
They could be distinguished by hostname (except in test environments where they're all on the same host -- in that case I have no solution to offer). Maybe that is not as strong a check as we would like.

There were two cases I had in mind that might justify supporting CA certs here:
1. Since the helpers need TLS server certificates as well, and currently use the same web server to serve both requests both from other helpers and external requests, it might be convenient for the helper to be able to obtain one cert from a public CA and use that for both client and server auth.
2. Having some private CA that issues certificates to all the helpers. This seems quite dubious.

I am happy to drop the support for using CA certs here. As you note, there is a testing gap that ought to be closed if we keep it.

Reply:
I'll leave the question of retaining code to your judgment, with the only stipulation being that if code is in the tree, it is tested somehow.

The question of a private CA doesn't really fit with my expectations here.  My general sense is that with mutually distrustful helpers (an important goal of ours), we don't want to create a single entity that can issue for all of them.  So that even pushes against the idea of having any CA-based system on the off-chance that two helpers ultimately choose the same CA and create the potential for collusion risk via that CA.

What I think is far more likely for client certs is that they are self-signed and sent around manually at the formation of the helper network and when their operational practices dictate.

Reply:
Oh, I should have added: this code could be useful still.

If the helpers jointly operate a CA to authenticate report collectors, that might be an OK way to manage client authentication for requests.  We're not building that now, but it seems like you could have a system like that (each helper could have its own CA. say, with all helpers recognizing the CA so that they can authenticate uploads.

I don't know if that is better than the usual token-based schemes; business needs will ultimately have to influence this somewhat.